src/models/chart-model/index.js

```diff
@@ -92,33 +103,100 @@ export default function createChartModel({
     previousConstraints: undefined,
     updateWithSettings: undefined,
     constraintsHaveChanged: undefined,
+    progressive: false,
   };
 
-  function updatePartial() {
-    meta.updateWithSettings = false;
-    requestAnimationFrame(() => {
-      // TODO: cancel requests as well to optimize???
-      // const startTime = Date.now();
-      trendLinesService.update();
+  let dataPages = [];
+  let enableDataPagesExtract = true;
+  let timer = null;
+
+  function extractDataPages() {
+    if (enableDataPagesExtract) {
```

Comment:
It would feel better if this could be handled inside the `renderProgressive` function instead. To determine wether the `extractDataPages` should be called or not.

Reply:
Fixed!